internal/nvmeof/controller/controllerserver.go

```diff
@@ -312,6 +385,17 @@ func validateCreateVolumeRequest(req *csi.CreateVolumeRequest) error {
 	if err != nil {
 		return fmt.Errorf("invalid listeners parameter: %w", err)
 	}
+	// Validate QoS parameters - cannot mix RBD and NVMe-oF QoS
+	hasRBDQoS := rbd.HasQoSParams(params)
+	nvmeofQoS, err := parseQoSParameters(params)
+	if err != nil {
+		return fmt.Errorf("invalid NVMe-oF QoS parameters: %w", err)
+	}
+	hasNVMeoFQoS := (nvmeofQoS != nil)
+
+	if hasRBDQoS && hasNVMeoFQoS {
+		return errors.New("cannot specify both RBD and NVMe-oF QoS parameters")
+	}
```

Comment:
These should be MutableParameters not the parameters during the CreateVolume isnt it?

Reply:
yes we should move to the mutable as per the CSI spec. lets open a tracker for it for nbd QOS as well.

Reply:
Yes thats correct expectation from me as well.